### PR TITLE
drivers/sx127x: remove ZTIMER_USEC dependency

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description

This PR removes the `ZTIMER_USEC` dependency of `sx127x`.
Timers here are used to ensure that reset pin timings are respected and to wait for the device to be ready when it's powered on for the first time. All this delays are expressed as minimum delay values, so It's possible to round them up to a millisecs scale.

This reduces ROM and some RAM from:
```
   text	  data	   bss	   dec	   hex	filename
  35260	   540	  4216	 40016	  9c50	/home/jialamos/Development/RIOT/tests/driver_sx127x/bin/b-l072z-lrwan1/tests_driver_sx127x.elf
```
to
```
   text	  data	   bss	   dec	   hex	filename
  34708	   496	  4208	 39412	  99f4	/home/jialamos/Development/RIOT/tests/driver_sx127x/bin/b-l072z-lrwan1/tests_driver_sx127x.elf
```

### Testing procedure
Check that applications depending on this driver still work (e.g `tests/driver_sx127x`, `examples/lorawan`, `examples/gnrc_lorawan`).

### Issues/PRs references
None so far, but the idea came from a discussion during the RIOT summit.

### CODE REVIEW

File: drivers/sx127x/sx127x.c [hunk 1]
```diff
@@ -42,20 +42,20 @@
...
+   timers, we round it to 1 ms */
+#define SX127X_MANUAL_RESET_SIGNAL_LEN_MS       (1U)
 
 /* After triggering a manual reset the device needs at least 5 ms to become
-   ready before interacting with it. To ensure this value is big enough even
-   with an inaccurate clock source, an additional 10 % error margin is added. */
-#define SX127X_MANUAL_RESET_WAIT_FOR_READY_US   (5500U)
+   ready before interacting with it. We round up to 6 ms in case the clock
+   source is not accurate enough */
+#define SX127X_MANUAL_RESET_WAIT_FOR_READY_MS   (6U)
```
Comment (kaspar030) [inline drivers/sx127x/sx127x.c:52]: I think the rounding up would not be needed, ztimer guarantees *at least* semantics. Not waiting long enough would be a ztimer bug. But I guess that 1ms extra wait doesn't hurt either...

**Comment by kaspar030** [comment]
I don't have a device to test. The change seems simple enough, so @jia200x I'd ACK if you did test this.

**Comment by kaspar030** [comment]
> This reduces ROM and some RAM from:

@MichelRottleuthner should we use this as datapoint for explicit timer clock selection?

File: drivers/sx127x/sx127x.c (hunk 1 above)
Comment (jia200x) [reply drivers/sx127x/sx127x.c:52]: Yeah, but I can imagine is the clock source is drifting forward the actual timer fires before the expected value, even though the semantics of ztimer guarantees "at least" semantics. Or I'm missing something?

**Comment by jia200x** [comment]
> I don't have a device to test. The change seems simple enough, so @jia200x I'd ACK if you did test this.

I will test it now with GNRC LoRaWAN. Write back in some minutes

File: drivers/sx127x/sx127x.c (hunk 1 above)
Comment (kaspar030) [reply drivers/sx127x/sx127x.c:52]: or did you mean the clock source of the sx127x? then it makes sense of course

Comment (jia200x) [reply drivers/sx127x/sx127x.c:52]: yes, I meant the clock source :). Ok, so I guess we can leave it as it is.

**Comment by jia200x** [comment]
Tested:
```
2020-09-16 12:03:54,130 #  send 4 "This is RIOT!"
2020-09-16 12:03:58,014 # PKTDUMP: data received:
2020-09-16 12:03:58,019 # ~~ SNIP  0 - size:   5 byte, type: NETTYPE_LORAWAN (1)
2020-09-16 12:03:58,023 # 00000000  AA  BB  CC  DD  EE
2020-09-16 12:03:58,027 # ~~ SNIP  1 - size:   1 byte, type: NETTYPE_UNDEF (0)
2020-09-16 12:03:58,028 # 00000000  02
2020-09-16 12:03:58,033 # ~~ SNIP  2 - size:   8 byte, type: NETTYPE_UNDEF (0)
2020-09-16 12:03:58,037 # 00000000  60  F5  51  01  26  20  00  00
2020-09-16 12:03:58,041 # ~~ PKT    -  3 snips, total size:  14 byte
2020-09-16 12:03:58,043 # Successfully sent packet
```

(Sent an uplink and also received a downlink here. So everything should work)

**Review by kaspar030 — APPROVED** [review]
ACK.

**Comment by MichelRottleuthner** [comment]
> @MichelRottleuthner should we use this as datapoint for explicit timer clock selection?

Sure! IMO every bit of information that shows how specific solutions affect relevant metrics is very valuable.
I guess the high-level timer RDM would be the right place to put that?

**Comment by kaspar030** [comment]
mh, this exposes a whole in ztimer's auto config. if rtt is not build, ztimer msec doesn't know on what to base itself. lemme try to fix.

**Comment by fjmolinas** [comment]
> mh, this exposes a whole in ztimer's auto config. if rtt is not build, ztimer msec doesn't know on what to base itself. lemme try to fix.

would something like this do?

```
ifneq (,$(filter ztimer_msec,$(USEMODULE)))
  USEMODULE += ztimer
  ifeq (,$(filter ztimer_periph_rtt,$(USEMODULE)))
    USEMODULE += ztimer_usec
  endif
endif
```

**Comment by kaspar030** [comment]
> would something like this do?

I think so.

**Comment by jia200x** [comment]
should I provide a PR with the fix?

**Comment by kaspar030** [comment]
> should I provide a PR with the fix?

that would be awesome!

**Comment by fjmolinas** [comment]
Can you rebase @jia200x ?

**Comment by jia200x** [comment]
rebased!

**Comment by fjmolinas** [comment]
@jia200x can you sneak in this patch:

```diff
diff --git a/cpu/fe310/include/periph_cpu.h b/cpu/fe310/include/periph_cpu.h
index d2593b1af0..6ce2c99660 100644
--- a/cpu/fe310/include/periph_cpu.h
+++ b/cpu/fe310/include/periph_cpu.h
@@ -167,12 +167,8 @@ typedef struct {
 #define RTT_MIN_FREQUENCY   (1U)                    /* in Hz */
 
 #ifndef RTT_FREQUENCY
-#ifdef MODULE_PERIPH_RTC
-#define RTT_FREQUENCY       (RTT_MIN_FREQUENCY)     /* in Hz */
-#else
 #define RTT_FREQUENCY       (RTT_MAX_FREQUENCY)     /* in Hz */
 #endif
-#endif
 
 /**
  * @brief   Initialization of the clock
```

**Comment by fjmolinas** [comment]
> @jia200x can you sneak in this patch:
> 
> ```diff
> diff --git a/cpu/fe310/include/periph_cpu.h b/cpu/fe310/include/periph_cpu.h
> index d2593b1af0..6ce2c99660 100644
> --- a/cpu/fe310/include/periph_cpu.h
> +++ b/cpu/fe310/include/periph_cpu.h
> @@ -167,12 +167,8 @@ typedef struct {
>  #define RTT_MIN_FREQUENCY   (1U)                    /* in Hz */
>  
>  #ifndef RTT_FREQUENCY
> -#ifdef MODULE_PERIPH_RTC
> -#define RTT_FREQUENCY       (RTT_MIN_FREQUENCY)     /* in Hz */
> -#else
>  #define RTT_FREQUENCY       (RTT_MAX_FREQUENCY)     /* in Hz */
>  #endif
> -#endif
>  
>  /**
>   * @brief   Initialization of the clock
> diff --git a/sys/ztimer/Makefile.include b/sys/ztimer/Makefile.include
> index 57c795bc82..6640f20b00 100644
> --- a/sys/ztimer/Makefile.include
> +++ b/sys/ztimer/Makefile.include
> @@ -8,7 +8,7 @@ endif
>  # might not be the most optimized for conversion guarantees that ztimer_periph_rtt
>  # will have a capable backend.
>  ifneq (,$(filter ztimer_periph_rtt,$(USEMODULE)))
> -  ifneq (,$(filter stm32 nrf52 sam% kinetis efm32,$(CPU)))
> +  ifneq (,$(filter stm32 nrf52 sam% kinetis efm32 fe310,$(CPU)))
>      RTT_FREQUENCY ?= RTT_MAX_FREQUENCY
>      CFLAGS += -DRTT_FREQUENCY=$(RTT_FREQUENCY)
>    endif
> ```

That will fix the failing build, please put them in two separate commits.

**Comment by jia200x** [comment]
sure! let me sneak it in. Thanks!

**Comment by jia200x** [comment]
done!

**Review by fjmolinas — APPROVED** [review]
ACK, if murdock is green please merge.

**Comment by fjmolinas** [comment]
@jia200x there seem to be an error with Kconfig modeling for the test, can you take a look?

**Comment by leandrolanzieri** [comment]
> @jia200x there seem to be an error with Kconfig modeling for the test, can you take a look?

Found it. `config MODULE_SX127X` is selecting the timer backend for ztimer, which is not needed with these changes I guess?

**Comment by fjmolinas** [comment]
ping @jia200x

**Comment by fjmolinas** [comment]
@leandrolanzieri I pushed forced the missing stuff, do I mind giving it a look so I don't ACK a change I pushed myself?

**Comment by fjmolinas** [comment]
> @leandrolanzieri I pushed forced the missing stuff, do I mind giving it a look so I don't ACK a change I pushed myself?

Assuming Murdock is green with this

**Comment by jia200x** [comment]
@fjmolinas thanks! I was right now on it but saw you were faster. It should be fine now I guess

**Comment by fjmolinas** [comment]
> @fjmolinas thanks! I was right now on it but saw you were faster. It should be fine now I guess

I guess if you are fine with my force push we can consider this OK.

**Comment by fjmolinas** [comment]
@jia200x hit the green button if that is the case!

**Comment by jia200x** [comment]
thanks for the review!

**Pull request merged**